ons-spark/spark-app/spark_application_and_ui.md

```diff
@@ -0,0 +1,429 @@
+```
+
+```{code-tab} r R
+
+config <- yaml::yaml.load_file("ons-spark/config.yaml")
+
+rescue <- sparklyr::spark_read_parquet(sc, config$rescue_path)
+
+print(paste0("Number of partitions: ", sparklyr::sdf_num_partitions(rescue)))
+
+```
+````
+
+```plaintext
+Number of partitions:  2
+```
+Next we'll carry out some basic processes to see how this translates into jobs, stages and tasks in the Spark UI. We will  group the incident costs by creating a new column called `cost_group` containing three groups `small`, `medium` and `large`.
+````{tabs}
+```{code-tab} py
+rescue = (
+    rescue.withColumn("cost_group",
+                      F.when(F.col("total_cost")<300, "small")
+                      .when((F.col("total_cost")>=300) & (F.col("total_cost")<900), "medium")
+                      .when(F.col("total_cost")>=1000, "large")
+                      .otherwise(None)
+                     )
+)
+
+rescue.select("animal_group", "description", "total_cost", "cost_group").show(7, truncate=80)
+```
+
+```{code-tab} r R
+
+rescue <- rescue %>% dplyr::mutate(
+    cost_group=dplyr::case_when(total_cost<300 ~ "small",
+                                total_cost>=300 & total_cost<900 ~ "medium",
+                                total_cost>=900 ~ "large",
+                                TRUE ~ NULL)
+)
+
+
+rescue %>% 
+    dplyr::select("animal_group", "description", "total_cost", "cost_group") %>%
+    head(7)
+
+```
+````
+
+```plaintext
++------------+----------------------------------------------------------------+----------+----------+
+|animal_group|                                                     description|total_cost|cost_group|
++------------+----------------------------------------------------------------+----------+----------+
+|         Cat|                                         CAT TRAPPED IN BASEMENT|     290.0|     small|
+|       Horse|                                           HORSE TRAPPED IN GATE|     590.0|    medium|
+|        Bird|PIGEON WITH WING IMAPLED ON SHARP IMPLEMENT  UNDER A BRIDGE NEAR|     326.0|    medium|
+|         Cat|                          ASSIST RSPCA WITH CAT STUCK ON CHIMNEY|     295.0|     small|
+|         Dog|                                       DOG FALLEN INTO THE CANAL|     260.0|     small|
+|        Deer|                                          DEER STUCK IN RAILINGS|     520.0|    medium|
+|        Deer|                                           DEER TRAPPED IN FENCE|     260.0|     small|
++------------+----------------------------------------------------------------+----------+----------+
+only showing top 7 rows
+```
+To produce the above `.show()`/`head()` Spark didn't need to process all of the partitions, there are enough rows on one partition to create the output so Spark only processed one partition. If we want Spark to process all of the partitions we need to call an action that involves all rows of the DataFrame, such as `.count()`.
```

Comment:
I'd put _To produce the above output_ rather than .show/head.

At the end I would either put _such as getting the row count_ or add sdf_nrow() after count().